lib/puppet/util/windows/registry.rb

```diff
@@ -65,6 +65,30 @@ def values(key)
       vals
     end
 
+    # Retrieve a set of values from a registry key given their names
+    # Value names listed but not found in the registry will not be added to the
+    # resultant Hashtable
+    #
+    # @param key [RegistryKey] An open handle to a Registry Key
+    # @param names [String[]] An array of names of registry values to return if they exist
+    # @return [Hashtable<String, Object>] A hashtable of all of the found values in the registry key
+    def values_by_name(key, names)
+      vals = {}
+      names.each do |name|
+        type, data = nil, nil
```

Comment:
Only worth assigning `data = nil`

Reply:
Turns out we don't need that either as `read` defaults its result to `nil` anyhow.... and if it's not returning a value, its throwing an exception we ignore.

Will update.